Tests/EventListener/RequestListenerTest.php

```diff
@@ -64,7 +64,10 @@ public function setUp()
 
         $this->mobileDetector = $this->getMockBuilder('SunCat\MobileDetectBundle\DeviceDetector\MobileDetector')->disableOriginalConstructor()->getMock();
         $this->deviceView = $this->getMockBuilder('SunCat\MobileDetectBundle\Helper\DeviceView')->disableOriginalConstructor()->getMock();
-        $this->router = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Routing\Router')->disableOriginalConstructor()->getMock();
+        $this->router = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Routing\Router')
+            ->disableOriginalConstructor()
+            ->setMethods(array('getRouteCollection'))
```

Comment:
Newer versions of the Routing component can optionally use the ExpressionLanguage component for more advanced conditions. However, as that feature is optional the ExpressionLanguage component is not installed when executing the tests and PHPUnit's mock library fails to mock methods where one of the arguments' type hint refers to a non-existing class or interface.